include/keyword_vpd_parser.hpp

```diff
@@ -88,6 +101,19 @@ class KeywordVpdParser : public ParserInterface
      */
     void checkNextBytesValidity(uint8_t numberOfBytes);
 
+    /**
+     * @brief API to iterate through the VPD vector to find the given keyword.
+     *
+     * This API iterates through VPD vector using m_vpdIterator and finds the
+     * given keyword. m_vpdIterator points to the keyword name if find is
```

Comment:
>And also we can't parse() or populateVpdMap() on this object as 'm_vpdIterator' is moved.

why can't you move again?

Reply:
the API name is findKeyword(), getter API's shouldn't modify anything.

> why can't you move again?
is this is a suggestion  or asking we can move it. ?

Reply:
This isn't a getter API. this just checks if the keyword is found or not. since we have m_iterator as a member variable which is common to any member of the class, this API uses it to iterate through the vector.

Remember findKeyword() is not modifying the content of the vector. It is using the iterator variable to iterate through the vector. If m_iterator shouldn't be modified, then give me the purpose of having it as a non-const class member.

Also this API acts as a common API to perform both read and write keyword. PR for write keyword https://github.com/ibm-openbmc/openpower-vpd-parser/pull/459

Reply:
>why can't you move again?
is this is a suggestion or asking we can move it. ?

Am telling you, if you wish to perform parse/populateVPDMap - you can move the iterator wherever you want to.